protocol/bc/transaction.go

```diff
@@ -110,6 +140,73 @@ func (tx *TxData) Value() (driver.Value, error) {
 	return b.Bytes(), nil
 }
 
+// WriteTo writes tx to w.
+func (tx *TxData) WriteTo(w io.Writer) (int64, error) {
+	ew := errors.NewWriter(w)
+	tx.writeTo(ew, serRequired)
+	return ew.Written(), ew.Err()
+}
+
+func (tx *TxData) writeTo(w io.Writer, serflags byte) error {
```

Comment:
With respect to `main`? Only in terms of comments and error wrapping - and the version on `main` is better. Updated.